controllers/kokumetricsconfig_controller.go

```diff
@@ -347,23 +428,52 @@ func setAuthentication(r *MetricsConfigReconciler, authConfig *crhchttp.AuthConf
 		}
 		return err
 	} else {
-		// No authentication secret name set when using basic auth
+		// No authentication secret name set when using basic or service-account auth
 		cr.Status.Authentication.AuthenticationCredentialsFound = &falseDef
-		err := fmt.Errorf("no authentication secret name set when using basic auth")
+		err := fmt.Errorf("no authentication secret name set when using basic or service-account auth")
 		cr.Status.Authentication.AuthErrorMessage = err.Error()
 		cr.Status.Authentication.ValidBasicAuth = &falseDef
 		return err
 	}
 }
 
-func validateCredentials(r *MetricsConfigReconciler, handler *sources.SourceHandler, cr *metricscfgv1beta1.MetricsConfig, cycle int64) error {
+func (r *MetricsConfigReconciler) validateCredentials(ctx context.Context, handler *sources.SourceHandler, cr *metricscfgv1beta1.MetricsConfig, cycle int64) error {
 	log := log.WithName("validateCredentials")
 
 	if cr.Spec.Authentication.AuthType == metricscfgv1beta1.Token {
 		// no need to validate token auth
 		return nil
 	}
 
+	// Service-account authentication check
+	if cr.Spec.Authentication.AuthType == metricscfgv1beta1.ServiceAccount {
+		if err := handler.Auth.GetAccessToken(ctx, cr.Spec.Authentication.TokenURL); err != nil {
+			var specificErrMsg string
+
+			if authErr, ok := err.(*crhchttp.AuthError); ok {
+				defaultMsg := "Invalid client credentials provided. Correct the client-id / client-secret"
+
+				switch {
+
+				case strings.Contains(authErr.ErrorType, "invalid_client"):
+					specificErrMsg = "Missing or invalid client credentials provided"
+				case strings.Contains(authErr.ErrorType, "unauthorized_client"):
+					specificErrMsg = "Missing or invalid client secret"
+				case strings.Contains(authErr.ErrorType, "unsupported_grant_type") || strings.Contains(authErr.Description, "grant_type"):
+					log.Error(authErr, "missing or invalid grant_type in request")
+					specificErrMsg = defaultMsg
+				default:
+					specificErrMsg = defaultMsg
+				}
+				log.Info(specificErrMsg, authErr)
+			}
+			errorMsg := fmt.Sprintf("%s in `%s`. Updated credentials will be re-verified during the next reconciliation.", specificErrMsg, cr.Spec.Authentication.AuthenticationSecretName)
+			cr.Status.Authentication.AuthErrorMessage = errorMsg
+			return err
```

Comment:
I think all of this is a bit excessive. I would just pass the error we get back from trying to get the token. The error_description we get back is sufficient to pass along. Like this:

```suggestion
			errorMsg := fmt.Sprintf("failed to acquire service-account token: %w" err)
			cr.Status.Authentication.AuthErrorMessage = errorMsg
			return err
```

Reply:
that sounds good to me. I've cleaned this up.